Sources/CombineRextensions/NavigationLink+Extensions.swift

```diff
@@ -32,7 +33,11 @@ extension NavigationLink {
                     viewModel.state[keyPath: pathToSelectedRowTag]
                 },
                 set: { row in
-                    viewModel.dispatch(dispatchActionOnTap(row ?? rowTag), from: .init(file: file, function: function, line: line, info: info))
```

Comment:
Dispatching action when opening is not enough, we have to dispatch action on close too.
We detect on close when the parameter in the Binding.set is nil.
If it's not nil, and it's the same as our hash, then it's our link being opened and we should dispatch the on open action.